squidpy/gr/_ripley.py

```diff
@@ -0,0 +1,245 @@
+    n_simulations: int = 100,
+    n_observations: int = 1000,
+    max_dist: np.float_ = None,
+    n_steps: int = 50,
+    seed: Optional[int] = None,
+    copy: bool = False,
+) -> Tuple[np.ndarray, np.ndarray, np.ndarray, np.ndarray]:
+    r"""
+    Calculate various Ripley's statistics for point processes.
+
+    According to the `'mode'` argument, it calculates either the following Ripley's statistics:
+    `{rp.F.s!r}`, `{rp.G.s!r}` or `{rp.L.s!r}` statistics.
+
+    `{rp.F.s!r}`, `{rp.G.s!r}` are defined as:
+
+    .. math::
+
+        F(t),G(t)=P( d_{{i,j}} \le t )
+
+    Where :math:`d_{{i,j}}` represents:
+
+        - distances to a random Spatial Poisson Point Process for `'F'`.
+        - distances to any other point of the dataset for `'G'`.
+
+    `{rp.L.s!r}` we first need to compute :math:`K(t)`, which is defined as:
+
+    .. math::
+
+        K(t) = \frac{{1}}{{\lambda}} \sum_{{i \ne j}} \frac{{I(d_{{i,j}}<t)}}{{n}}
+
+    and then we apply a variance-stabilizying transformation:
+
+    .. math::
+
+        L(t) = (\frac{{K(t)}}{{\pi}})^{{1/2}}
+
+    For reference, check out
+    `Wikipedia <https://en.wikipedia.org/wiki/Spatial_descriptive_statistics#Ripley's_K_and_L_functions>`_
+    or :cite:`Baddeley2015-lm`.
+
+    Parameters
+    ----------
+    %(adata)s
+    %(cluster_key)s
+    mode
+        Which Ripley's statistic to compute.
+    %(spatial_key)s
+    metric
+        Which metric to use for computing distances.
+    n_neigh
+        Number of neighborhoods to consider for the knn graph.
+    n_simulations
+        How many simulations to run for computing pvalues.
+    n_observations
+        How many observations to generate for the spatial poisson point process.
+    max_dist
+        Maximum distances for the support. If `None`, `max_dist=`:math:`\sqrt{{area \over 2}}`.
+    n_steps
+        Number of steps for the support.
+    %(seed)s
+    %(copy)s
+
+    Returns
+    -------
+    %(ripley_stat_returns)s
+    """
+    _assert_categorical_obs(adata, key=cluster_key)
+    _assert_spatial_basis(adata, key=spatial_key)
+    coordinates = adata.obsm[spatial_key]
+    clusters = adata.obs[cluster_key].values
+
+    mode = RipleyStat(mode)  # type: ignore[assignment]
+    if TYPE_CHECKING:
+        assert isinstance(mode, RipleyStat)
+    mode = mode.s
+
+    # prepare support
+    N = coordinates.shape[0]
+    hull = ConvexHull(coordinates)
+    area = hull.volume
+    if max_dist is None:
+        max_dist = (area / 2) ** 0.5
+    support = np.linspace(0, max_dist, n_steps)
+
+    # prepare labels
+    le = LabelEncoder().fit(clusters)
+    cluster_idx = le.transform(clusters)
+    obs_arr = np.empty((le.classes_.shape[0], n_steps))
+
+    start = logg.info(
+        f"Calculating Ripley's {mode} statistic for `{le.classes_.shape[0]}` clusters and `{n_simulations}` simulations"
+    )
+
+    for i in np.arange(np.max(cluster_idx) + 1):
+        coord_c = coordinates[cluster_idx == i, :]
+        if mode == "F":
```

Comment:
~~You're comparing an enum to string, can be dangerous if the impl. of `ModeEnum` changes in the future.
Please use `mode == RipleyStat.F`, etc.~~
My bad, you're just assigning `mode = mode.s` right after. However, I wouldn't do that and would just compare the enums, because if you change the value in the enum, PyCharm/your editor of preference won't most likely pick it up.